src/legacy/core_plugins/vis_type_vega/public/vega_view/vega_base_view.js

```diff
@@ -279,20 +278,14 @@ export class VegaBaseView {
    * @param {string} [index] as defined in Kibana, or default if missing
    */
   async removeFilterHandler(query, index) {
-    const $injector = await chrome.dangerouslyGetActiveInjector();
     const indexId = await this._findIndex(index);
     const filter = esFilters.buildQueryFilter(query, indexId);
 
-    // This is a workaround for the https://github.com/elastic/kibana/issues/18863
```

Comment:
I checked and the filter gets removed and the chart gets re-rendered correctly. However there are several other things broken around vega and its filter management that makes it practically impossible to use the feature right now. I'll open a separate issue for those.